karapace/schema_reader.py

```diff
@@ -308,10 +319,17 @@ def handle_messages(self) -> None:
                     if msg_keymode == KeyMode.DEPRECATED_KARAPACE:
                         self.key_formatter.set_keymode(KeyMode.DEPRECATED_KARAPACE)
 
+                def parse_value(raw_value: str) -> JsonObject | None:
+                    value = json_decode(raw_value)
+                    if not isinstance(value, dict):
+                        assert value == ""
+                        return None
+                    return value
+
                 value = None
                 if msg.value:
                     try:
-                        value = json_decode(msg.value)
+                        value = parse_value(msg.value)
```

Comment:
This is a balancing act between `test_regression_config_for_inexisting_object_should_not_throw` on the one hand that expects to be able to pass `value=""`, and `self.handle_msg()` that only seem to be accidentally resilient to string values in some special cases.

Perhaps the test should actually be passing `None`? It's hard to tell.

Reply:
While reading the changes I can see that e.g. pushing string or number can be decoded as valid json but same time cause probably odd exceptions from karapace code elsewhere. This change makes it a bit clearer if such unexpected situations happen.